### PR TITLE
fix the problem of  json output

### PR DESCRIPTION
**Pull request by d-d-up**
Signed-off-by: d-d-up <qhr6113@163.com>

**What this PR does / why we need it**:
Fixes: issue https://github.com/helm/helm/issues/11385
**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/registry/client.go [hunk 1]
```diff
@@ -416,8 +417,8 @@ func (c *Client) Pull(ref string, options ...PullOption) (*PullResult, error) {
 		}
 	}
 
-	fmt.Fprintf(c.out, "Pulled: %s\n", result.Ref)
-	fmt.Fprintf(c.out, "Digest: %s\n", result.Manifest.Digest)
+	fmt.Fprintf(os.Stderr, "Pulled: %s\n", result.Ref)
```
Comment (a-palchikov) [inline pkg/registry/client.go:420]: This kind of defeats the purpose of the `out` attribute and it does not solve the issue completely as below you can see another write to `c.out` (in this case it is a warning/error).
I would suggest to configure os.Stderr at the [root](https://github.com/helm/helm/blob/ce66412a723e4d89555dc67217607c6579ffcb21/cmd/helm/root.go#L158):

```go
		registry.ClientOptWriter(os.Stderr),
```

**Comment by yxxhero** [comment]
@d-d-up ping

**Comment by jouve** [comment]
as a side effect, this breaks [helmfile/helmfile](https://github.com/helmfile/helmfile/) template

**Comment by yxxhero** [comment]
close this PR, see: https://github.com/helm/helm/pull/11480

**Pull request closed without merging**